axelor-contract/src/main/java/com/axelor/apps/contract/service/ConsumptionLineServiceImpl.java

```diff
@@ -37,7 +37,9 @@ public ConsumptionLineServiceImpl(AppBaseService appBaseService) {
   @Override
   public ConsumptionLine fill(ConsumptionLine line, Product product) {
     Preconditions.checkNotNull(product, I18n.get(IExceptionMessage.CONTRACT_EMPTY_PRODUCT));
-    line.setLineDate(appBaseService.getTodayDate());
+    line.setLineDate(
+        appBaseService.getTodayDate(
+            line.getContractLine().getContractVersion().getContract().getCompany()));
```

Comment:
it would be safer to check for NPE (I think using optional.map will be useful here)